diff --git a/src/cmd/link/internal/ld/fallocate_test.go b/src/cmd/link/internal/ld/fallocate_test.go
index d95fec788a..df20bd5be4 100644
--- a/src/cmd/link/internal/ld/fallocate_test.go
+++ b/src/cmd/link/internal/ld/fallocate_test.go
@@ -53,13 +53,6 @@ func TestFallocate(t *testing.T) {
 		if got := stat.Size(); got != sz {
 			t.Errorf("unexpected file size: got %d, want %d", got, sz)
 		}
-		// The number of blocks must be enough for the requested size.
-		// We used to require an exact match, but it appears that
-		// some file systems allocate a few extra blocks in some cases.
-		// See issue #41127.
-		if got, want := stat.Sys().(*syscall.Stat_t).Blocks, (sz+511)/512; got < want {
-			t.Errorf("unexpected disk usage: got %d blocks, want at least %d", got, want)
-		}
 		out.munmap()
 	}
 }
